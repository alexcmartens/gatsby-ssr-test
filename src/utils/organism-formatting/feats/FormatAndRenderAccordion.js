import React from "react"
import { marked } from "marked"
import formatTheme from "../formatTheme"
import renderPaddingOrFullBleed from "../renderPaddingOrFullBleed"
import formatIcons from "../formatIcons"
import { Accordion } from "terra-one"

const FormatAndRenderAccordion = ({
  data,
  paddingClass,
  overrideTheme,
  nested,
}) => {
  const themeValue =
    overrideTheme?.split(" ").join("-").toLowerCase() ||
    data?.colorScheme?.split(" ").join("-").toLowerCase() ||
    "transparent-light-text"

  const items =
    data?.items?.map(item => {
      return {
        icon: formatIcons(item?.textItemIcon?.icon),
        headline: item?.headline || item?.fields?.headline,
        body: item?.body && (item?.body?.childMarkdownRemark?.html || marked?.parse(item?.body)),
      }
    }) || []
  const theme = formatTheme({ theme: themeValue })

  if (
    data?.colorScheme === "Blue" ||
    data?.colorScheme === "Dark Navy" ||
    data?.colorScheme === "Dark Gray"
  ) {
    theme.darkTheme = true
  }
  const disableIcon = data?.disableIcon
  const contentWidth = data?.contentWidth || false
  return renderPaddingOrFullBleed(
    <Accordion
      items={items}
      theme={theme}
      disableIcon={disableIcon}
      paddingClass={paddingClass}
    />,
    !nested ? contentWidth : contentWidth === false ? contentWidth : true,
    data?.backgroundColorPadding,
    true,
    nested
  )
}

export default FormatAndRenderAccordion
