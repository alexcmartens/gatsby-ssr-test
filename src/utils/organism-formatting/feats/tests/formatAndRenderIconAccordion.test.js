import React from "react"
import renderer from "react-test-renderer"
import FormatAndRenderAccordion from "../FormatAndRenderAccordion"

jest.mock("../../formatTheme")

describe("FormatAndRenderAccordion", () => {
  it("renders correctly", () => {
    const data = {
      theme: "White",
      items: [
        {
          icon: "ico-annotate-48px",
          headline: "Headline Lorem ipsum dolor sit amet",
          body: {
            childMarkdownRemark: {
              html: "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Blandit vitae   molestie aliquam iaculis eu et ultricies nec.</p>",
            },
          },
        },
      ],
      disableIcon: false,
    }

    const tree = renderer
      .create(<FormatAndRenderAccordion data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
