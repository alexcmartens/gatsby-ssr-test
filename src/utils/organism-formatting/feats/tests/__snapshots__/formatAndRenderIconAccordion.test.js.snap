// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormatAndRenderAccordion renders correctly 1`] = `
<div
  disableIcon={false}
  items={
    Array [
      Object {
        "body": "<p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Blandit vitae   molestie aliquam iaculis eu et ultricies nec.</p>",
        "headline": "Headline Lorem ipsum dolor sit amet",
        "icon": undefined,
      },
    ]
  }
  theme={
    Object {
      "accordionIconColor": "#0063A3",
      "backgroundColor": "#F8F8FB",
      "color": "#242D2F",
      "ctaOne": "primary--1",
      "ctaTwo": "secondary--1",
      "iconColor": "#252A2E",
      "theme": "theme-1",
    }
  }
/>
`;
